### DISCUSSION TITLE
"pack" extension type Documentation

### DISCUSSION THREAD

**Discussion by Dominic-Marcelino** [directus/directus#15001] [Feature Requests]
We recently had another discussion on discord about using multiple extension-types in one extension. 
The solution seems to be the `pack` extension-type.

As it's neither in the docs, nor supported by extensions-cli I'm curious for more information. 

* Is the pack-type already production-ready?
* If not: What is missing to finish the support?
* Are there any other resources / examples / docs on how it works and the code should be bundled?

Thanks a lot ☺️

**Comment by rijkvanzanten** [comment]
@nickrum any wisdom? 🧙🏻

**Reply by claytongulick** [reply]
This is exactly the issue that's driving us to use a forked internal version. Also the ability to use vanilla web components in extensions.

I addressed it with https://github.com/directus/directus/pull/14465

It didn't seem to drive a lot of interest so I haven't updated the PR for the recent releases, but I do have it all updated and merged in our production fork.

**Reply by Dominic-Marcelino** [reply]
Sorry to bring this up again, but are there any updates/ informations about this? 🤔😅

**Comment by nickrum** [comment]
> Is the pack-type already production-ready?

It should be, but it might not be doing what you are looking for. The `pack` type is exclusive to package extensions and acts as a "virtual" container for other package extensions. When you create an extension with `"type": "pack"` in `package.json` and you add other package extensions to the list of dependencies of this extension, Directus will load all extensions by traversing the list of dependencies of the `pack` extension.

**Reply by Dominic-Marcelino** [reply]
Thanks for clarification!

> and you add other package extensions to the list of dependencies of this extension

Does this only works with npm dependencies or can we also define somehow local dependencies within the directus options? 

E.g  something like `directus-extension-endpoint` from `./endoint/dist/index.js`and ``directus-extension-interface` from `./interface/dist/index.js`

**Reply by nickrum** [reply]
Right now it only works for npm dependencies / package extensions. But we are currently looking into different ways by which we can support this.

**Reply by Dominic-Marcelino** [reply]
Thanks! ☺️ Looking can't wait for the results.
To be honest it's a bit frustrating right now that we need to move all build files into the different folders, even if they belong to the same extension-set 😅

**Reply by egidiusmengelberg** [reply]
Yeah, really looking forward to this feature. I'm running directus in kubernetes and it's kind of a hassle to add all the different extension build files to their folders even though I'm only creating one new feature. It would be nice to have one extension that consists of multiple types.

Most of my extensions require migrations. But I'm currently manually building these in the UI because it's just faster than adding those using the command line.

**Comment by rijkvanzanten** [comment]
This has since been added
